pimcore/models/Asset.php

```diff
@@ -987,50 +992,55 @@ public function delete()
 
         \Pimcore::getEventDispatcher()->dispatch(AssetEvents::PRE_DELETE, new AssetEvent($this));
 
-        $this->closeStream();
+        try {
+            $this->closeStream();
 
-        // remove childs
-        if ($this->hasChildren()) {
-            foreach ($this->getChildren() as $child) {
-                $child->delete();
+            // remove childs
+            if ($this->hasChildren()) {
+                foreach ($this->getChildren() as $child) {
+                    $child->delete();
+                }
             }
-        }
 
-        // remove file on filesystem
-        $fullPath = $this->getRealFullPath();
-        if ($fullPath != '/..' && !strpos($fullPath, '/../') && $this->getKey() !== '.' && $this->getKey() !== '..') {
-            $this->deletePhysicalFile();
-        }
+            // remove file on filesystem
+            $fullPath = $this->getRealFullPath();
+            if ($fullPath != '/..' && !strpos($fullPath, '/../') && $this->getKey() !== '.' && $this->getKey() !== '..') {
+                $this->deletePhysicalFile();
+            }
 
-        $versions = $this->getVersions();
-        foreach ($versions as $version) {
-            $version->delete();
-        }
+            $versions = $this->getVersions();
+            foreach ($versions as $version) {
+                $version->delete();
+            }
 
-        // remove permissions
-        $this->getDao()->deleteAllPermissions();
+            // remove permissions
+            $this->getDao()->deleteAllPermissions();
 
-        // remove all properties
-        $this->getDao()->deleteAllProperties();
+            // remove all properties
+            $this->getDao()->deleteAllProperties();
 
-        // remove all metadata
-        $this->getDao()->deleteAllMetadata();
+            // remove all metadata
+            $this->getDao()->deleteAllMetadata();
 
-        // remove all tasks
-        $this->getDao()->deleteAllTasks();
+            // remove all tasks
+            $this->getDao()->deleteAllTasks();
 
-        // remove dependencies
-        $d = $this->getDependencies();
-        $d->cleanAllForElement($this);
+            // remove dependencies
+            $d = $this->getDependencies();
+            $d->cleanAllForElement($this);
 
-        // remove from resource
-        $this->getDao()->delete();
+            // remove from resource
+            $this->getDao()->delete();
 
-        // empty object cache
-        $this->clearDependentCache();
+            // empty asset cache
+            $this->clearDependentCache();
 
-        //set object to registry
-        \Pimcore\Cache\Runtime::set('asset_' . $this->getId(), null);
+            // clear asset from registry
+            \Pimcore\Cache\Runtime::set('asset_' . $this->getId(), null);
+        } catch (\Exception $e) {
+            \Pimcore::getEventDispatcher()->dispatch(AssetEvents::POST_DELETE_FAILURE, new AssetEvent($this));
```

Comment:
is there a throw $e missing at the end of the exception handler  ?

Reply:
we definitely still should throw the exceptions.

Reply:
Done :-) thanks for the catch 👍